### ISSUE TITLE
use "rich" for terminal rendering

### ISSUE THREAD

**Issue by dlax** [dalibo/pg_activity#261]
I'm thinking of replacing blessed with [rich](https://github.com/willmcgugan/rich) as a terminal rendering library. Rich is now mature and provides high level features that would be nice for us as we could drop a bunch of custom logic on our side. In particular:

* no more need to handle available vertical space to limit the number of processes rendered (the `limit()` decorator)
* automatic cropping or wrapping of long lines
* better handling of text split with terminal sequence (would allow to handle horizontal scrolling as suggested in #260 without loosing the formatting, or introducing a bunch of custom logic on our side) 
* native table rendering
* "layouts" allowing to split the screen into multiple areas (possibly with fixed size, and automatic overflow handling)
* live display, allowing to refresh only part of the ui

That would be a large refactoring. But, after having played a bit with it, I think it's worth as it will make UI development easier long term.

Thoughts?

**Comment by blogh** [comment]
That looks like a great idea. :)

**Comment by blogh** [comment]
It would be nice, if we could solve #222 during the revamp. (I noticed that if there is wrapping only the first line is colored).

**Comment by dlax** [comment]
One thing I'm not sure we can easily replace is the overlay confirmation 
dialog when the user requests process cancel or kill. It's possible to 
emit a confirmation message, similar to input(), with rich; but it's 
less fancy than the current widget.

**Comment by blogh** [comment]
The fact that the size of the columns is potentially changing at each refresh is a little distracting.
What do you think about this ?

**Comment by dlax** [comment]
> The fact that the size of the columns is potentially changing at each 
> refresh is a little distracting.
> What do you think about this ?

I agree. Either we can fix the size of columns. Or we could also just 
hide the table when there's no row to show. Needs more thinking in any case.
